components/gantry/oneaxis/oneaxis.go

```diff
@@ -167,6 +166,48 @@ func newOneAxis(ctx context.Context, deps resource.Dependencies, conf resource.C
 	return oAx, nil
 }
 
+func (g *oneAxis) Reconfigure(ctx context.Context, deps resource.Dependencies, conf resource.Config) error {
+	newConf, err := resource.NativeConfig[*Config](conf)
+	if err != nil {
+		return err
+	}
+
+	if g.lengthMm != newConf.LengthMm {
+		g.lengthMm = newConf.LengthMm
+	}
+	if g.rpm != newConf.GantryRPM {
+		g.rpm = newConf.GantryRPM
+	}
+	if g.mmPerRevolution != newConf.MmPerRevolution {
+		g.mmPerRevolution = newConf.MmPerRevolution
+	}
+
+	// Rerun homing if anything with the limit switch pins changes
+	if (len(g.limitSwitchPins) != len(newConf.LimitSwitchPins)) || (g.limitHigh != *newConf.LimitPinEnabled) {
```

Comment:
if both the limit switch changes and the motor name changes in a reconfigure, won't this run the homing sequence twice?

I'd rather do a check for all of these in a function that returns a bool (can be inline in the reconfigure function) and then use some smart logic to rehome once if any of these traits have changed at the end. 

It also makes it more readable. Something like
```go
doIRunHomeSequenceAgain := func(g.limitPins, g.motors) bool {
      switchesChanged := (len(g.limitSwitchPins) != len(newConf.LimitSwitchPins)) || (g.limitHigh != *newConf.LimitPinEnabled)
      
      var numSwitchesChanged = false
      for i, pin := range g.limitSwitchPins {
			if pin != newConf.LimitSwitchPins[i] { 
			   numSwitchesChanged = true
			}
			
	motorsChanged := g.motor.Name().ShortName() != newConf.Motor
       
       return motorsChanged || numSwitchesChanged || switchesChanged     
}
```

And in the main function
```go
     if  rehome := checker(g.limitSwitchPins, g.motor) {
           g.home()
      }
```

I'm sure you can make it better.

Reply:
About to commit the change I made that I think accomplishes this, but with Alan's suggested method. Let me know if you would rather I structure it this way!